library/core/src/ffi/c_str.rs

```diff
@@ -207,6 +213,26 @@ impl fmt::Display for FromBytesWithNulError {
     }
 }
 
+#[unstable(feature = "ub_checks", issue = "none")]
+impl Invariant for &CStr {
+    /**
+     * Safety invariant of a valid CStr:
+     * 1. An empty CStr should has a null byte.
+     * 2. A valid CStr should end with a null-terminator and contains
+     *    no intermediate null bytes.
+     */
+    fn is_safe(&self) -> bool {
+        let bytes: &[c_char] = &self.inner;
+        let len = bytes.len();
+
+        if bytes.is_empty() || bytes[len - 1] != 0 || bytes[..len-1].contains(&0) {
```

Comment:
Nit: can we change this to the condition of safety instead of unsafety? i.e.
```rust
!bytes.empty() && bytes[len - 1] == 0 && !bytes[..len-1].contains(&0)
```
We can also remove the `if`.